From 9539a9258d1af498bd9e94db74fa61c7ecd28047 Mon Sep 17 00:00:00 2001
From: Tiago Gaspar <tiagogaspar8@gmail.com>
Date: Wed, 22 Jun 2022 10:55:21 +0100
Subject: [PATCH] ath11k: enable ethernet offload by default

Enable ath11k ethernet encapsulation/decapsulation offload
by default. This improves wireless performance without
requiring user configuration.

Signed-off-by: Tiago Gaspar <tiagogaspar8@gmail.com>
---
 package/kernel/mac80211/ath.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/package/kernel/mac80211/ath.mk b/package/kernel/mac80211/ath.mk
index 7e1dfa1f5b28d..91566a1d2bbaf 100644
--- a/package/kernel/mac80211/ath.mk
+++ b/package/kernel/mac80211/ath.mk
@@ -307,6 +307,9 @@ define KernelPackage/ath11k
   +kmod-crypto-michael-mic +ATH11K_THERMAL:kmod-hwmon-core +ATH11K_THERMAL:kmod-thermal
   FILES:=$(PKG_BUILD_DIR)/drivers/soc/qcom/qmi_helpers.ko \
   $(PKG_BUILD_DIR)/drivers/net/wireless/ath/ath11k/ath11k.ko
+  AUTOLOAD:=$(call AutoProbe,ath11k)
+  # Enable encapsulation/decapsulation offload by default
+  MODPARAMS.ath11k:=frame_mode=2
 endef
 
 define KernelPackage/ath11k/description
